Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -57,15 +57,31 @@ bool LogFileReader::readAll(const QString& fileName)
  * that the conversion from string to double is sucessful.*/
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
+    bool check = true;
```

Comment:
I like the optimisation of your code, however it could be more readable and explicit about whats happening, instead of using 'check' for both the voltage and the current, how could you use an extra variable and do a final check at the end?